bin/BuildPackages.sh

```diff
@@ -21,55 +50,82 @@ set -e
 # an idea what to do for a complete installation of GAP.
 
 
-if [ $# -eq 0 ]
-  then
-    echo "Assuming default GAP location: ../.."
-    GAPDIR=../..
-  else
-    echo "Using GAP location: $1"
-    GAPDIR="$1"
+# Is someone trying to run us from inside the 'bin' directory?
+if [[ -f gapicon.bmp ]]
+then
+  error "This script must be run from inside the pkg directory" \
+        "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# Is someone trying to run us from inside the 'bin' directory?
-if [ -f gapicon.bmp ]
-  then
-    echo "This script must be run from inside the pkg directory"
-    echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
-    exit 1
+# CURDIR
+CURDIR="$(pwd)"
+
+# GAPDIR
+if [[ "$#" -ge 1 ]]
+then
+  case "$1" in
+    --with-gaproot=*)
+      GAPDIR="$(sed 's|--with-gaproot=||' <<< $1)"
+      shift
+    ;;
+
+    *)
+      cd ..
+      GAPDIR="$(pwd)"
+      cd "$CURDIR"
+    ;;
```

Comment:
Yes, from now on the code
````
cd /my/custom/pkg
$GAPDIR/bin/BuildPackages.sh $GAPDIR
````
does not work. If you do not start the first argument by ````--with-gaproot````, then it is considered to be a package name. Now I added a warning about this into the error message if ````$GAPDIR```` is not found properly.

(To be honest, I did not want to write a very sophisticated check on whether the argument is a proper gap directory or a package or something else. I felt that this was an easier way.)

In any case, I applied all your suggestions here.